src/http/api/views/account.posts.view.ts

```diff
@@ -0,0 +1,667 @@
+import {
+    Activity,
+    CollectionPage,
+    isActor,
+    lookupObject,
+} from '@fedify/fedify';
+import type { Account } from 'account/account.entity';
+import { getAccountHandle } from 'account/utils';
+import type { FedifyContextFactory } from 'activitypub/fedify-context.factory';
+import { type Result, error, ok } from 'core/result';
+import { sanitizeHtml } from 'helpers/html';
+import { isUri } from 'helpers/uri';
+import type { Knex } from 'knex';
+import { PostType } from 'post/post.entity';
+import type { PostDTO } from '../types';
+
+export type GetPostsError =
+    | 'invalid-next-parameter'
+    | 'error-getting-outbox'
+    | 'no-page-found'
+    | 'not-an-actor';
+
+interface BaseGetProfileDataResultRow {
+    post_id: number;
+    post_type: PostType;
+    post_title: string | null;
+    post_excerpt: string | null;
+    post_content: string | null;
+    post_url: string;
+    post_image_url: string | null;
+    post_published_at: Date;
+    post_like_count: number;
+    post_liked_by_current_user: 0 | 1;
+    post_reply_count: number;
+    post_reading_time_minutes: number;
+    post_repost_count: number;
+    post_reposted_by_current_user: 0 | 1;
+    post_ap_id: string;
+    post_attachments: {
+        type: string | null;
+        mediaType: string | null;
+        name: string | null;
+        url: string;
+    }[];
+    author_id: number;
+    author_name: string | null;
+    author_username: string;
+    author_url: string | null;
+    author_avatar_url: string | null;
+}
+
+interface GetProfileDataResultRowReposted extends BaseGetProfileDataResultRow {
+    reposter_id: number;
+    reposter_name: string | null;
+    reposter_username: string;
+    reposter_url: string | null;
+    reposter_avatar_url: string | null;
+}
+
+interface GetProfileDataResultRowWithoutReposted
+    extends BaseGetProfileDataResultRow {
+    reposter_id: null;
+    reposter_name: null;
+    reposter_username: null;
+    reposter_url: null;
+    reposter_avatar_url: null;
+}
+
+export type GetProfileDataResultRow =
+    | GetProfileDataResultRowReposted
+    | GetProfileDataResultRowWithoutReposted;
+
+export interface AccountPosts {
+    results: PostDTO[];
+    nextCursor: string | null;
+}
+
+export class AccountPostsView {
+    constructor(
+        private readonly db: Knex,
+        private readonly fedifyContextFactory: FedifyContextFactory,
+    ) {}
+
+    async getPostsByHandle(
+        handle: string,
+        account: Account,
+        defaultAccount: Account,
+        limit: number,
+        cursor: string | null,
+    ): Promise<Result<AccountPosts, GetPostsError>> {
+        if (!account.id || !defaultAccount.id) {
+            throw new Error('Account or default account not found');
+        }
+
+        //If we found the account in our db and it's an internal account, do an internal lookup
+        if (account?.isInternal) {
```

Comment:
I don't think `account` is optional here

Reply:
Now it's optional. So I kept it like this